torch/lib/THD/master_worker/master/Master.cpp

```diff
@@ -4,29 +4,50 @@
 #include "../worker/Worker.h"
 #include "../../process_group/General.hpp"
 
+#include <string>
+#include <thread>
 
 namespace thd {
-
 namespace master {
 
 std::unique_ptr<MasterCommandChannel> masterCommandChannel;
+std::thread masterErrorThread;
```

Comment:
I think it's cleaner to make this thread a member of MasterCommandChannel. It will start it once it's initialized.